packages/react-liveness/src/components/FaceLivenessDetector/shared/MatchIndicator.tsx

```diff
@@ -38,3 +38,5 @@ export const MatchIndicator: React.FC<MatchIndicatorProps> = ({
     </div>
   );
 };
+
+export const MemoizedMatchIndicator = React.memo(MatchIndicator);
```

Comment:
Could be completely off base here, but it's unclear to me how wrapping `MatchIndicator` with `React.memo` is improving performance since we are only passing `percentage` to it from `LivenessCameraModule`. Thinking the usage of `Math.ceil` [here](https://github.com/aws-amplify/amplify-ui/pull/4122/files#diff-5d58d3114c36274fae912867705fd03fd3a44874ea478e657d41ce9224c520dbR255) might be enough since it reduces the amount of re-renders by only passing whole numbers to `MatchIndicator`?

Reply:
I was thinking the same thing but despite that, whenever the parent component re-rendered it would also cause this component to re-render. Adding the `React.memo` to it made it so that this child component would not re-render even though the parent had. We have a separate task for tracking performance of the parent component in this case

Reply:
Makes sense. Maybe can add a comment here with a concise explanation for future maintainers?

Reply:
Just curious, why not directly use `React.useMemo` inside `LivenessCameraModule.tsx`?

Reply:
no particular reason, just decided to do it this way since the change was based around this package. Do we prefer doing it that way instead?

Reply:
FWIW think it makes sense to memo where it is needed, ideally the memory used for the memoization frees up once `LiveCameraModule` unmounts, and also removes the mental overhead of whether to use `MemoizedMatchIndicator` or `MatchIndicator` for future devs (know it's unlikely `MatchIndicator` will ever be reused but just saying)

Reply:
Sure, i can update it to just be useMemo in place!

Reply:
Think in general we prefer `React.useMemo`:
- we usually use `React.useMemo`: [search](https://github.com/search?q=repo%3Aaws-amplify%2Famplify-ui+React.memo&type=code)
- By using `useMemo`, the parent component has more control over how it's memoized
- Other components that uses the same component aren't affected

That said `React.memo` works perfectly fine in this case because we are just memoizing the whole component, but I think it'll be nicer if we stick with `useMemo` for the reasons above.

Reply:
Ended up switching to use `React.useMemo` but realized this slightly increased the number of renders because the memoized component will forcibly re-render when the LivenessCameraModule component is re-rendered (something else we plan to optimize separately).

Because `react.memo` was used at the HOC level it will reduce re-renders regardless if it's parent component re-renders.

Anyways what do you guys think?

Reply:
Started wondering abt that actually since the HOC is applied outside the scope of `LivenessCameraModule`. IMO wrapping with `React.memo` is fine, would prefer it happens in _LivenessCameraModule.tsx_ with an explanatory comment. Ideally we should be able to isolate the rendering issues in `LivenessCameraModule` by breaking it in to smaller components and remove the `memo` in the future

Reply:
Added the react.memo and a comment explainer